Modules/posixmodule.c

```diff
@@ -12118,53 +12149,77 @@ os_mknod_impl(PyObject *module, path_t *path, int mode, dev_t device,
 
 #ifdef HAVE_DEVICE_MACROS
 /*[clinic input]
-os.major -> unsigned_int
+os.major
 
     device: dev_t
     /
 
 Extracts a device major number from a raw device number.
 [clinic start generated code]*/
 
-static unsigned int
+static PyObject *
 os_major_impl(PyObject *module, dev_t device)
-/*[clinic end generated code: output=5b3b2589bafb498e input=1e16a4d30c4d4462]*/
+/*[clinic end generated code: output=4071ffee17647891 input=b1a0a14ec9448229]*/
 {
-    return major(device);
+    unsigned int result = major(device);
+#ifdef NODEV
```

Comment:
Can you add a helper function to convert an "unsigned int" device number to a Python object? os.minor() uses the same code.